File: app/src/main/java/cdreyfus/xebia_henri_potier/models/Slice.java
package cdreyfus.xebia_henri_potier.models;

public class Slice extends CommercialOffer{
    private int value;
    private int sliceValue;

    public Slice(int value, int sliceValue) {
        this.type="slice";
        this.value = value;
        this.sliceValue = sliceValue;
    }

    public int getSliceValue() {
        return sliceValue;
    }

    public void setSliceValue(int sliceValue) {
        this.sliceValue = sliceValue;
    }

    public int getValue() {
        return value;
    }

    public void setValue(int value) {
        this.value = value;
    }

    public float applySlice(float basketValue){
        return basketValue - ((int) basketValue / sliceValue) * value;
    }

    @Override
    public float applyOffer(float basketValue) {
        return applySlice(basketValue);
    }
}

File: app/src/main/java/cdreyfus/xebia_henri_potier/basket/promotion/Slice.kt
package cdreyfus.xebia_henri_potier.basket.promotion

class Slice(var value: Int, var sliceValue: Int) : CommercialOffer() {

    init {
        this.type = "slice"
    }

    fun applySlice(basketValue: Float): Float {
        return basketValue - basketValue.toInt() / sliceValue * value
    }

    override fun applyOffer(basketValue: Float): Float {
        return applySlice(basketValue)
    }
}
